src/main/java/org/digitalstack/chess/ChessBoard.java

```diff
@@ -12,10 +14,40 @@ public ChessBoard() {
     }
 
     public void add(Pawn pawn, int xCoordinate, int yCoordinate, PieceColor pieceColor) {
-        throw new UnsupportedOperationException("Need to implement ChessBoard.add()");
+    	int numberOfWhites = 0;
+        int numberOfBlacks = 0;
+    	if(isLegalBoardPosition(xCoordinate, yCoordinate) && pieces[xCoordinate][yCoordinate] == null 
```

Comment:
Negate this condition and fail fast. You will remove one layer of imbrication this way.